experimental/PropertyDDS/packages/property-properties/src/properties/boolProperty.js

```diff
@@ -6,26 +6,31 @@
  * @fileoverview Definition of the BooleanProperty class
  */
 
-const ValueProperty = require('./valueProperty');
-const _castFunctors = require('./primitiveTypeCasts');
+const { ValueProperty } = require('./valueProperty');
+const { _castFunctors } = require('./primitiveTypeCasts');
 
 /**
  * A primitive property for a boolean value
- * @param {Object=} in_params - the parameters
- * @constructor
- * @protected
- * @extends property-properties.ValueProperty
- * @alias property-properties.BoolProperty
- * @category Value Properties
  */
-var BoolProperty = function (in_params) {
-    ValueProperty.call(this, in_params);
-    // default for this property type is 'false'
-    this._data = false;
-};
-BoolProperty.prototype = Object.create(ValueProperty.prototype);
+export class BoolProperty extends ValueProperty {
+
+    /**
+     * @param {Object=} in_params - the parameters
+     * @constructor
+     * @protected
+     * @extends property-properties.ValueProperty
+     * @alias property-properties.BoolProperty
+     * @category Value Properties
+     */
+    constructor(in_params) {
+        super({ typeid: 'Bool', ...in_params });
+        // default for this property type is 'false'
+        this._data = false;
+    };
+
+    _castFunctor = _castFunctors.Boolean;
```

Comment:
I think you should do `_castFunctors.Boolean.bind(this)` in the constructor if you want to map it correctly to es6. 
Probably performance wise it's better to modify prototype directly (keep it as is), because .bind(..) could be expensive if it's called on every instantiation.